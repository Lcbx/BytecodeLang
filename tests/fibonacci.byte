def n = 10

def a = 0
def b = 1
def c = 0
while n > 0
	c = a + b
	b = a
	a = c
	n = n -1

# the Nth number of the suite
a