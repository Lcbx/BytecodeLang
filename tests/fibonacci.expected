tokenize_Out:
NAME: def, NAME: n, OP: =, INT: 10, NAME: def, NAME: a, OP: =, INT: 0, NAME: def, NAME: b, OP: =, INT: 1, NAME: def, NAME: c, OP: =, INT: 0, NAME: while, NAME: n, OP: >, INT: 0, TABS: 1, NAME: c, OP: =, NAME: a, OP: +, NAME: b, TABS: 1, NAME: b, OP: =, NAME: a, TABS: 1, NAME: a, OP: =, NAME: c, TABS: 1, NAME: n, OP: =, NAME: n, OP: -, INT: 1, NAME: n, OP: =, INT: 10, OP: +, INT: 5, EOF: line 12
__SECTION__
tokenize_Err:

__SECTION__
compile_Out:
0 errors
42 instructions [4, 10, 4, 0, 4, 1, 4, 0, 9, 0, 4, 0, 19, 14, 26, 0, 11, 9, 1, 9, 2, 21, 10, 3, 9, 1, 10, 2, 9, 3, 10, 1, 9, 0, 4, 1, 22, 10, 0, 12, 224, 255]
__SECTION__
compile_Err:

__SECTION__
simulation_Out:
0    	OP_INT1 => 10 --
2    	OP_INT1 => 0 --
4    	OP_INT1 => 1 --
6    	OP_INT1 => 0 --
8    	OP_LOAD => var0 (10) --
10   	OP_INT1 => 0 --
12   	OP_GT => True --
13   	OP_JUMP_IF_FALSE => !jump 26 --
16   	OP_POP => True --
17   	OP_LOAD => var1 (0) --
19   	OP_LOAD => var2 (1) --
21   	OP_ADD => 1 --
22   	OP_STORE => var3 = 1 (was 0) --
24   	OP_LOAD => var1 (0) --
26   	OP_STORE => var2 = 0 (was 1) --
28   	OP_LOAD => var3 (1) --
30   	OP_STORE => var1 = 1 (was 0) --
32   	OP_LOAD => var0 (10) --
34   	OP_INT1 => 1 --
36   	OP_SUB => 9 --
37   	OP_STORE => var0 = 9 (was 10) --
39   	OP_JUMP => jumped -32 --
10   	OP_INT1 => 0 --
12   	OP_GT => True --
13   	OP_JUMP_IF_FALSE => !jump 26 --
16   	OP_POP => True --
17   	OP_LOAD => var1 (1) --
19   	OP_LOAD => var2 (0) --
21   	OP_ADD => 1 --
22   	OP_STORE => var3 = 1 (was 1) --
24   	OP_LOAD => var1 (1) --
26   	OP_STORE => var2 = 1 (was 0) --
28   	OP_LOAD => var3 (1) --
30   	OP_STORE => var1 = 1 (was 1) --
32   	OP_LOAD => var0 (9) --
34   	OP_INT1 => 1 --
36   	OP_SUB => 8 --
37   	OP_STORE => var0 = 8 (was 9) --
39   	OP_JUMP => jumped -32 --
10   	OP_INT1 => 0 --
12   	OP_GT => True --
13   	OP_JUMP_IF_FALSE => !jump 26 --
16   	OP_POP => True --
17   	OP_LOAD => var1 (1) --
19   	OP_LOAD => var2 (1) --
21   	OP_ADD => 2 --
22   	OP_STORE => var3 = 2 (was 1) --
24   	OP_LOAD => var1 (1) --
26   	OP_STORE => var2 = 1 (was 1) --
28   	OP_LOAD => var3 (2) --
30   	OP_STORE => var1 = 2 (was 1) --
32   	OP_LOAD => var0 (8) --
34   	OP_INT1 => 1 --
36   	OP_SUB => 7 --
37   	OP_STORE => var0 = 7 (was 8) --
39   	OP_JUMP => jumped -32 --
10   	OP_INT1 => 0 --
12   	OP_GT => True --
13   	OP_JUMP_IF_FALSE => !jump 26 --
16   	OP_POP => True --
17   	OP_LOAD => var1 (2) --
19   	OP_LOAD => var2 (1) --
21   	OP_ADD => 3 --
22   	OP_STORE => var3 = 3 (was 2) --
24   	OP_LOAD => var1 (2) --
26   	OP_STORE => var2 = 2 (was 1) --
28   	OP_LOAD => var3 (3) --
30   	OP_STORE => var1 = 3 (was 2) --
32   	OP_LOAD => var0 (7) --
34   	OP_INT1 => 1 --
36   	OP_SUB => 6 --
37   	OP_STORE => var0 = 6 (was 7) --
39   	OP_JUMP => jumped -32 --
10   	OP_INT1 => 0 --
12   	OP_GT => True --
13   	OP_JUMP_IF_FALSE => !jump 26 --
16   	OP_POP => True --
17   	OP_LOAD => var1 (3) --
19   	OP_LOAD => var2 (2) --
21   	OP_ADD => 5 --
22   	OP_STORE => var3 = 5 (was 3) --
24   	OP_LOAD => var1 (3) --
26   	OP_STORE => var2 = 3 (was 2) --
28   	OP_LOAD => var3 (5) --
30   	OP_STORE => var1 = 5 (was 3) --
32   	OP_LOAD => var0 (6) --
34   	OP_INT1 => 1 --
36   	OP_SUB => 5 --
37   	OP_STORE => var0 = 5 (was 6) --
39   	OP_JUMP => jumped -32 --
10   	OP_INT1 => 0 --
12   	OP_GT => True --
13   	OP_JUMP_IF_FALSE => !jump 26 --
16   	OP_POP => True --
17   	OP_LOAD => var1 (5) --
19   	OP_LOAD => var2 (3) --
21   	OP_ADD => 8 --
22   	OP_STORE => var3 = 8 (was 5) --
24   	OP_LOAD => var1 (5) --
26   	OP_STORE => var2 = 5 (was 3) --
28   	OP_LOAD => var3 (8) --
30   	OP_STORE => var1 = 8 (was 5) --
32   	OP_LOAD => var0 (5) --
34   	OP_INT1 => 1 --
36   	OP_SUB => 4 --
37   	OP_STORE => var0 = 4 (was 5) --
39   	OP_JUMP => jumped -32 --
10   	OP_INT1 => 0 --
12   	OP_GT => True --
13   	OP_JUMP_IF_FALSE => !jump 26 --
16   	OP_POP => True --
17   	OP_LOAD => var1 (8) --
19   	OP_LOAD => var2 (5) --
21   	OP_ADD => 13 --
22   	OP_STORE => var3 = 13 (was 8) --
24   	OP_LOAD => var1 (8) --
26   	OP_STORE => var2 = 8 (was 5) --
28   	OP_LOAD => var3 (13) --
30   	OP_STORE => var1 = 13 (was 8) --
32   	OP_LOAD => var0 (4) --
34   	OP_INT1 => 1 --
36   	OP_SUB => 3 --
37   	OP_STORE => var0 = 3 (was 4) --
39   	OP_JUMP => jumped -32 --
10   	OP_INT1 => 0 --
12   	OP_GT => True --
13   	OP_JUMP_IF_FALSE => !jump 26 --
16   	OP_POP => True --
17   	OP_LOAD => var1 (13) --
19   	OP_LOAD => var2 (8) --
21   	OP_ADD => 21 --
22   	OP_STORE => var3 = 21 (was 13) --
24   	OP_LOAD => var1 (13) --
26   	OP_STORE => var2 = 13 (was 8) --
28   	OP_LOAD => var3 (21) --
30   	OP_STORE => var1 = 21 (was 13) --
32   	OP_LOAD => var0 (3) --
34   	OP_INT1 => 1 --
36   	OP_SUB => 2 --
37   	OP_STORE => var0 = 2 (was 3) --
39   	OP_JUMP => jumped -32 --
10   	OP_INT1 => 0 --
12   	OP_GT => True --
13   	OP_JUMP_IF_FALSE => !jump 26 --
16   	OP_POP => True --
17   	OP_LOAD => var1 (21) --
19   	OP_LOAD => var2 (13) --
21   	OP_ADD => 34 --
22   	OP_STORE => var3 = 34 (was 21) --
24   	OP_LOAD => var1 (21) --
26   	OP_STORE => var2 = 21 (was 13) --
28   	OP_LOAD => var3 (34) --
30   	OP_STORE => var1 = 34 (was 21) --
32   	OP_LOAD => var0 (2) --
34   	OP_INT1 => 1 --
36   	OP_SUB => 1 --
37   	OP_STORE => var0 = 1 (was 2) --
39   	OP_JUMP => jumped -32 --
10   	OP_INT1 => 0 --
12   	OP_GT => True --
13   	OP_JUMP_IF_FALSE => !jump 26 --
16   	OP_POP => True --
17   	OP_LOAD => var1 (34) --
19   	OP_LOAD => var2 (21) --
21   	OP_ADD => 55 --
22   	OP_STORE => var3 = 55 (was 34) --
24   	OP_LOAD => var1 (34) --
26   	OP_STORE => var2 = 34 (was 21) --
28   	OP_LOAD => var3 (55) --
30   	OP_STORE => var1 = 55 (was 34) --
32   	OP_LOAD => var0 (1) --
34   	OP_INT1 => 1 --
36   	OP_SUB => 0 --
37   	OP_STORE => var0 = 0 (was 1) --
39   	OP_JUMP => jumped -32 --
10   	OP_INT1 => 0 --
12   	OP_GT => False --
13   	OP_JUMP_IF_FALSE => jump 26 --
__SECTION__
simulation_Err:
